src/pybind/mgr/cephadm/module.py

```diff
@@ -2443,8 +2439,15 @@ def generate_grafana_ds_config(hosts: List[str]) -> str:
             )
 
         prom_services = [ps.hostname for ps in self.cache.get_daemons_by_service('prometheus')]
-        cert, pkey = create_self_signed_cert('Ceph', 'cephadm')
-        config_file = json.dumps({
+        if cert_fname and pkey_fname:
+            with open(cert_fname, 'r') as f:
+                cert = f.read()
+            with open(pkey_fname, 'r') as f:
+                pkey = f.read()
+        else:
+            cert, pkey = create_self_signed_cert('Ceph', 'cephadm')
```

Comment:
hmm, i wonder if instead of creating a new cert each time the daemon is deployed (configured?) and plumbing this through the orchestrator interface, perhaps it would be better to 
- pass the daemon name to the config generator function(s)
- at config time, pull the cert from config-key
- at create time, verify the config-key key exists.  if not, generate a self-signed one.
- if the user wants a specific cert, they can set it in config-key before deploying the service

@sebastian-philipp what do you think?

Reply:
I hope it's how it's supposed to be now. May you check?